lib/config/env.ts

```diff
@@ -86,14 +87,52 @@ export function getConfig(env: NodeJS.ProcessEnv): GlobalConfig {
     });
   }
 
-  if (env.DOCKER_USERNAME && env.DOCKER_PASSWORD) {
-    config.hostRules.push({
-      hostType: datasourceDocker.id,
-      username: env.DOCKER_USERNAME,
-      password: env.DOCKER_PASSWORD,
-    });
+  const datasources = new Set(getDatasourceList());
+  const fields = ['token', 'username', 'password'];
+
+  const hostRules: HostRule[] = [];
+
+  for (const envName of Object.keys(env).sort()) {
+    // Double underscore __ is used in place of hyphen -
+    const splitEnv = envName.toLowerCase().replace('__', '-').split('_');
+    const hostType = splitEnv.shift();
+    if (datasources.has(hostType)) {
+      const suffix = splitEnv.pop();
+      if (fields.includes(suffix)) {
+        let matchHost: string;
+        const rule: HostRule = {};
+        rule[suffix] = env[envName];
+        if (splitEnv.length === 0) {
+          // host-less rule
+        } else if (splitEnv.length === 1) {
+          logger.warn(`Cannot parse ${envName} env`);
+          continue; // eslint-disable-line no-continue
+        } else {
+          matchHost = splitEnv.join('.');
+        }
+        const existingRule = hostRules.find(
+          (hr) => hr.hostType === hostType && hr.matchHost === matchHost
+        );
+        if (existingRule) {
+          // Add current field to existing rule
+          existingRule[suffix] = env[envName];
+        } else {
+          // Create a new rule
+          const newRule: HostRule = {
+            hostType,
+          };
+          if (matchHost) {
+            newRule.matchHost = matchHost;
+          }
+          newRule[suffix] = env[envName];
+          hostRules.push(newRule);
+        }
+      }
+    }
   }
 
+  config.hostRules = [...config.hostRules, ...hostRules];
+
   // These env vars are deprecated and deleted to make sure they're not used
   const unsupportedEnv = [
```

Comment:
Everything that is old is new again 😂 

If you really wanted to, you could write a compatibility layer for these 😉